ch2driver/pytpcc/drivers/nestcollectionsdriver.py

```diff
@@ -795,21 +433,24 @@ def __init__(self, ddl, clientId, TAFlag="T",
                 txnQueries = CH2PP_TXN_QUERIES
             for txn, queries in txnQueries.items():
                 for query, statement in queries.items():
+                    stmt = statement
+                    if self.schema == constants.CH2_DRIVER_SCHEMA["CH2P"]:
+                        stmt = re.sub("default:bench.ch2pp.", "default:bench.ch2p.", statement)
```

Comment:
Also, in the regex pattern we should escape the `.`s so that we actually match literal `.` characters instead of anything.

i.e.
```
stmt = re.sub("default:bench\.ch2pp\.", "default:bench.ch2p.", statement)
```

Reply:
An alternative approach to all of this could also be to supply a `query_context` when submitting N1QL queries (see https://docs.couchbase.com/cloud/n1ql/n1ql-manage/query-settings.html#query_context, and also https://docs.couchbase.com/cloud/n1ql/n1ql-language-reference/prepare.html#query-context for some info about query contexts for prepared statements)

With this approach, all the transaction statement definitions can just use not-fully-qualified table names like `orders`, and the `query_context` can be set to `constants.CH2_NAMESPACE + ":" + constants.CH2_BUCKET + "." self.schema`.

Reply:
According to the second link above, prepared statements are also "stored relative to the current query context", which seems to mean that if we prepare queries while providing a specific query context like `default:bench.ch2pp` then we can prepare another query with the same name with a different query context and they won't interfere.

> A prepared statement is created and stored relative to the current query context. You can create multiple prepared statements with the same name, each stored relative to a different query context. This enables you to run multiple instances of the same application against different datasets.
> 
> To execute a prepared statement, the query context must be the same as it was when the prepared statement was created; otherwise the prepared statement will not be found.

Reply:
We have CH2_TXN_QUERIES for CH2, and CH2PP_TXN_QUERIES for both CH2P and CH2PP, so CH2 and CH2PP are taken care of correctly. 
We handle it like this:
if self.schema == constants.CH2_DRIVER_SCHEMA["CH2"]:
                txnQueries = CH2_TXN_QUERIES
            else:
                txnQueries = CH2PP_TXN_QUERIES
            for txn, queries in txnQueries.items():
                for query, statement in queries.items():
                    stmt = statement
                    if self.schema == constants.CH2_DRIVER_SCHEMA["CH2P"]:
                        stmt = re.sub("default:bench.ch2pp.", "default:bench.ch2p.", statement)

Instead of duplicating and adding another dictionary for CH2P, I replace ch2pp with ch2p.

Reply:
Made the change to escape the .

Reply:
Agreed. This was inherited from the nestcollectionsdriver written by the query service team for the transactions part, so that is the history of this.
Your suggestion to set the correct query context is fantastic, but I am not sure how to set this via the rest api. If you can find some way to do it, we can add it later, but for now, let us pass on this if ok with you.